### PR TITLE
Jest test

### PR DESCRIPTION
**Pull request by NeilRashbrook**

### CODE REVIEW

File: protocol.js [hunk 1]
```diff
@@ -36,19 +36,19 @@ export default class JPCWebSocket extends JPCProtocol {
...
     return new Promise((resolve, reject) => {
-      server.on("connection", async webSocket => {
+      this._server.on("listening", () => resolve(this._server.address().port));
+      this._server.on("connection", async webSocket => {
         try {
           await this.init(webSocket);
-          resolve();
         } catch (ex) {
-          reject(ex);
+          console.log(ex);
```
Comment (benbucksch) [inline protocol.js]: Why not reject? I think we really should tell the caller. Otherwise this hangs?

(If a log, then it would be `console.error(ex)`)

Comment (benbucksch) [reply protocol.js]:
```suggestion
          reject(ex);
```

**Review by benbucksch — APPROVED** [review]

File: protocol.js (hunk 1 above)
Comment (NeilRashbrook) [reply protocol.js]: The promise was resolved on listen, so this does nothing at all.

Comment (benbucksch) [reply protocol.js]: Understood. "connection" event is called on every connect of a client, i.e. multiple times, so we cannot pass it on to the caller.

But we can change it into `console.error(ex);`.

Comment (NeilRashbrook) [reply protocol.js]: Sorry, yes, I missed that bit, `console.error` would certainly have been correct here.

**Pull request merged**